waiter/src/waiter/process_request.clj

```diff
@@ -496,7 +497,11 @@
           (try
             (confirm-live-connection-without-abort)
             (let [{:keys [service-id service-description] :as descriptor} (request->descriptor-fn request)
-                  {:strs [metric-group]} service-description]
+                  {:strs [metric-group]} service-description
+                  ^DateTime request-time (t/now)]
+              (->> (utils/date-to-str request-time utils/formatter-rfc822)
+                   (add-debug-header-into-response! "x-waiter-request-timestamp"))
```

Comment:
Name suggestion: `x-waiter-request-date` (a la the HTTP `Date` http header) or `x-waiter-request-datetime` or `x-waiter-request-time`